loleaflet/src/layer/tile/CanvasTileLayer.js

```diff
@@ -5472,12 +5472,144 @@ L.CanvasTileLayer = L.Layer.extend({
 		this._map.fire('docsize', {x: scrollPixelLimits.x, y: scrollPixelLimits.y, extraSize: extraSize});
 	},
 
+	// In file based view, all parts may be drawn, not only selected part.
+	_getTileInfoForFileBasedView: function (currentX, currentY, zoom, partHeightPixels, queue) {
+		var yMin = currentY;
+		var yMax = yMin + this._tileSize;
+
+		var parts = [];
+		for (var i = 0; i < this._parts; i++) {
+			var partMin = i * partHeightPixels;
+			var partMax = (i + 1) * partHeightPixels;
+			if ((yMin >= partMin && yMin <= partMax) || (yMax >= partMin && yMax <= partMax)) {
```

Comment:
What if one or more part(s) can fit completely in a tile at a low enough zoom ?

Reply:
Then partHeightPixels will be lower too. It is a dynamic variable. But really good catch by the way:) I'll check it again to see if everything is ok. Thanks for all the comments, i am taking them into account, some of them are now in co-6-4, i will send some follow-ups when i'm ready for the details like LPoly-CPoly convertions.